Homework 21

timepritchard opened the pull request:
I am stuck at clearing the text in the playlist name - I tried the .clear and then I tried the method shown in the class but neither works

Berichicko commented:
> I am stuck at clearing the text in the playlist name - I tried the .clear and then I tried the method shown in the class but neither works

Hi

You got this problem, please correct it.

![image](https://github.com/user-attachments/assets/908d14d4-be90-408a-93c1-4ed512e54f42)

HINT

```
public void enterNewPlaylist(String playlist) {
    WebElement playlistInput = findElement(newPlaylist);
    
    // Select text in the input field (Ctrl + A on Windows, Command + A on MacBook)
    playlistInput.sendKeys(Keys.CONTROL, "a");
    playlistInput.sendKeys(Keys.COMMAND, "a");
    
    // Delete the selected text (Delete on Windows, BackSpace on MacBook)
    playlistInput.sendKeys(Keys.DELETE);
    playlistInput.sendKeys(Keys.BACK_SPACE);
    
    // Input new text
    playlistInput.sendKeys(playlist);
    
    // Press the "Enter" key to confirm
    playlistInput.sendKeys(Keys.RETURN);
}
```
![image](https://github.com/mlordlie/A49-QA-Automation-Framework/assets/104761482/a486fed8-91a1-4412-bf4f-51b6e48f27ac)

or

![image](https://github.com/user-attachments/assets/ede6d11d-73c9-4001-a40e-3bf6c71629a2)

TestPro-Support commented:
Hello,

I have reviewed your homework:
![image](https://github.com/user-attachments/assets/025ff434-06cb-4209-8649-d6f6414c7c88)

Please add the missing element specifier in your XPath expression:

Sample Syntax: 
//element[@name='name']

Example:
//input[@name='name']

timepritchard commented:
Hi, I have now updated it - but even with your recommendation it didn't work to find the element with the xpath - instead I changed to the css locator and this worked - why wouldn't it work using xpath?

TestPro-Support commented:
Hello,

Upon review, it appears your XPath expression for the clearPlaylist() method is missing the tag:
  (By.xpath("[@name='name']")));
![image](https://github.com/user-attachments/assets/0593b851-0b14-4a1e-9fc8-bfabe519bc90)

**XPath Syntax:**

**XPath with Tag:** To select an <input> element with a specific attribute, you should include the tag in your XPath. For example: //input[@name='name'] — This selects <input> elements with a name attribute of 'name'.

**XPath with Wildcard:** If you want to select an element without specifying the tag, you can use the * wildcard, such as 
//*[@name='name'] — This selects any element with a name attribute of 'name'.

Please ensure to include the appropriate tag in your XPath to accurately locate the element.
Feel free to reach out if you have any questions or need further assistance.
